edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -117,14 +128,27 @@ protected internal async Task<T> Execute<T>(Func<Task<T>> func, string operation
             try
             {
                 Events.ExecutingOperation(operation, this.ManagementUri.ToString());
-                T result = await ExecuteWithRetry(func, r => Events.RetryingOperation(operation, this.ManagementUri.ToString(), r), this.transientErrorDetectionStrategy)
+                T result = await ExecuteWithRetry(
+                    func,
+                    (r) => Events.RetryingOperation(operation, this.ManagementUri.ToString(), r),
+                    this.transientErrorDetectionStrategy)
                     .TimeoutAfter(this.operationTimeout);
                 Events.SuccessfullyExecutedOperation(operation, this.ManagementUri.ToString());
+                this.failedSocketCount = 0;
                 return result;
             }
             catch (Exception ex)
             {
                 Events.ErrorExecutingOperation(ex, operation, this.ManagementUri.ToString());
+                if (this.IsModuleManagementSocketStale(ex))
+                {
+                    this.failedSocketCount++;
+                    if (this.failedSocketCount >= MaxRetryCount)
+                    {
+                        Environment.Exit(0);
```

Comment:
Add Log.Error with specific event id and message.

Reply:
Sure. I can use `110` error code that it was throwing. I thought that we capture and handle the error correctly; hence, it should just shutdown with error code `0`

Reply:
` Log.Error` is done by `Events.ErrorExecutingOperation`